### PR TITLE
GlobalContext: Don't require manual sync with SectionManager.c

### PR DESCRIPTION
**Pull request by riidefi**

### CODE REVIEW

File: payload/game/ui/GlobalContext.cc [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 #include <sp/CourseDatabase.hh>
 
+// Imported by SectionManager.c
+extern "C" s16 GlobalContext_SizeInBytes;
```
Comment (GnomedDev) [inline payload/game/ui/GlobalContext.cc:10]: Is this `extern` needed?

File: payload/game/ui/SectionManager.c [hunk 2]
```diff
@@ -1,5 +1,14 @@
 #include <Common.h>
 
 void SectionManager_init(void *this);
+extern s16 GlobalContext_SizeInBytes;
 
-PATCH_S16(SectionManager_init, 0x8e, 0x510 + sizeof(u32) * (1 + 11) + 0x88);
+__attribute__((section("patches"))) const Patch SectionManagerPatch = {
```
Comment (GnomedDev) [inline payload/game/ui/SectionManager.c:6]: Comment to explain the need for a manual macro expansion?

**Review by GnomedDev — APPROVED** [review]
LGTM, just some minor nitpicks if you care.

File: payload/game/ui/SectionManager.c (hunk 2 above)
Comment (riidefi) [reply payload/game/ui/SectionManager.c:6]: Good point

File: payload/game/ui/GlobalContext.cc (hunk 1 above)
Comment (riidefi) [reply payload/game/ui/GlobalContext.cc:10]: In practice no--can't hurt to be explicit about C linkage though.

**Pull request merged**